aws-datastore/src/main/java/com/amplifyframework/datastore/DataStoreConfiguration.java

```diff
@@ -0,0 +1,340 @@
+        this.dataStoreConflictHandler = dataStoreConflictHandler;
+        this.syncIntervalInMinutes = syncIntervalInMinutes;
+        this.syncMaxRecords = syncMaxRecords;
+        this.syncPageSize = syncPageSize;
+    }
+
+    /**
+     * Begin building a new instance of {@link DataStoreConfiguration}.
+     * @return A new builder instance
+     */
+    @NonNull
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Begin building a new instance of {@link DataStoreConfiguration} by reading DataStore
+     * settings from the config file.
+     * @param pluginJson DataStore plugin configuration from amplicationconfiguration.json
+     * @param userProvidedConfiguration An instance of {@DataStoreConfiguration} with settings specified by the user
+     *                                  which will be used as overrides.
+     * @return A new builder instance
+     * @throws DataStoreException exception thrown if there's an unexpected configuration key or
+     * an invalid configuration value
+     */
+    @NonNull
+    public static Builder builder(@NonNull JSONObject pluginJson,
+                                  @NonNull DataStoreConfiguration userProvidedConfiguration) throws DataStoreException {
+        return builder(pluginJson)
+            .syncPageSize(userProvidedConfiguration.getSyncPageSize())
+            .syncMaxRecords(userProvidedConfiguration.getSyncMaxRecords())
+            .syncIntervalMs(userProvidedConfiguration.getSyncIntervalInMinutes())
+            .dataStoreErrorHandler(userProvidedConfiguration.getDataStoreErrorHandler())
+            .dataStoreConflictHandler(userProvidedConfiguration.getDataStoreConflictHandler());
+    }
+
+    /**
+     * Begin building a new instance of {@link DataStoreConfiguration}.
+     * @param pluginJson The dataStore configuration as a JSONObject
+     * @return A new builder instance
+     * @throws DataStoreException exception thrown if there's an unexpected configuration key or
+     * an invalid configuration value
+     */
+    @NonNull
+    public static Builder builder(@NonNull JSONObject pluginJson) throws DataStoreException {
+        final Iterator<String> jsonKeys = pluginJson.keys();
+        Builder builder = new Builder();
+        while (jsonKeys.hasNext()) {
+            final String keyString = jsonKeys.next();
+            final ConfigKey configKey;
+            try {
+                configKey = ConfigKey.fromString(keyString);
+            } catch (IllegalArgumentException noSuchConfigKey) {
+                throw new DataStoreException(
+                        "Saw unexpected config key: " + keyString,
+                        "Make sure your amplifyconfiguration.json is valid."
+                );
+            }
+            try {
+                switch (configKey) {
+                    case SYNC_INTERVAL:
+                        builder.syncIntervalMs(pluginJson.getLong(ConfigKey.SYNC_INTERVAL.toString()));
```

Comment:
Since this one is user-exposed, I think we want this work in minutes, here. _Inside_ the builder, convert to ms.